src/server/controller/plugin.ts

```diff
@@ -0,0 +1,139 @@
+import { PostgresGateway } from "../gateways/index";
+import { Trade } from "../types";
+
+export default class Controller {
+  db: PostgresGateway;
+
+  constructor() {
+    this.db = new PostgresGateway();
+  }
+
+  public async write(
+    body: Trade
+  ): Promise<{
+    rows: Array<{ id: string }>;
+  }> {
+    try {
+      const id = await this.db.write(body);
+      return id;
+    } catch (err) {
+      this.db.logger.log("error", "Error:", err);
```

Comment:
**Comment**: It should not be the responsibility of the db to handle errors for the controller.

Reply:
Updated controllers to extend gateway. This way, controllers will have access to it's own logger.